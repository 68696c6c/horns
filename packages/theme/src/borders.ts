import { Size } from './sizes'
import { Side } from './utils'

export enum BorderStyle {
  None = 'none',
  Hidden = 'hidden',
  Dotted = 'dotted',
  Dashed = 'dashed',
  Solid = 'solid',
  Double = 'double',
  Groove = 'groove',
  Ridge = 'ridge',
  Inset = 'inset',
  Outset = 'outset',
  Initial = 'initial',
  Inherit = 'inherit',
}

export type BorderProperties = {
  width?: Size
  style?: BorderStyle
}

export type BordersConfig = {
  [key in Side]?: BorderProperties
}
