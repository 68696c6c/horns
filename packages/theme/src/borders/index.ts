import { SideSizes, SideSizesConfig, Size } from '../sizes'
import { configToSides, Sides, SidesConfig } from '../utils'

export enum BorderStyle {
  None = 'none',
  Hidden = 'hidden',
  Dotted = 'dotted',
  Dashed = 'dashed',
  Solid = 'solid',
  Double = 'double',
  Groove = 'groove',
  Ridge = 'ridge',
  Inset = 'inset',
  Outset = 'outset',
  Initial = 'initial',
  Inherit = 'inherit',
}

// i.e. Size
export type BorderProperties = {
  width?: Size
  style?: BorderStyle
}

// i.e. isSize
export const isBorderProperties = (
  tbd?: SideBordersConfig | BorderProperties,
): tbd is BorderProperties => {
  if (typeof tbd === 'undefined') {
    return false
  }
  const hasStyle = typeof (tbd as BorderProperties).style === 'string'
  const hasWidth = typeof (tbd as BorderProperties).width === 'string'
  return hasStyle || hasWidth
}

// i.e. SideSizes
export type SideBorders = Sides<BorderProperties>
export type SideBorderStyles = Sides<BorderStyle>

// i.e. defaultSideSizes
const defaultBorderWidths: SideSizes = {
  top: Size.None,
  bottom: Size.None,
  left: Size.None,
  right: Size.None,
}
const defaultBorderStyles: SideBorderStyles = {
  top: BorderStyle.None,
  bottom: BorderStyle.None,
  left: BorderStyle.None,
  right: BorderStyle.None,
}
const defaultSideBorders: SideBorders = {
  top: {},
  bottom: {},
  left: {},
  right: {},
}

// i.e. SideSizesConfig
export type SideBordersConfig = SidesConfig<BorderProperties>
type SideBorderStylesConfig = SidesConfig<BorderStyle>

type Extracted = {
  widths: SideSizesConfig
  styles: SideBorderStylesConfig
}

const extract = (value: SideBordersConfig | BorderProperties): Extracted => {
  const props: SideBordersConfig = isBorderProperties(value)
    ? { all: value }
    : value
  const { all, x, y, top, bottom, left, right } = props
  const widths: SideSizesConfig = {}
  const styles: SideBorderStylesConfig = {}
  if (all) {
    widths.all = all.width
    styles.all = all.style
  }
  if (x) {
    widths.x = x.width
    styles.x = x.style
  }
  if (y) {
    widths.y = y.width
    styles.y = y.style
  }
  if (top) {
    widths.top = top.width
    styles.top = top.style
  }
  if (bottom) {
    widths.bottom = bottom.width
    styles.bottom = bottom.style
  }
  if (left) {
    widths.left = left.width
    styles.left = left.style
  }
  if (right) {
    widths.right = right.width
    styles.right = right.style
  }
  return { widths, styles }
}

// i.e. evalSideSizesConfig
export const evalSideBordersConfig = (
  defaults: SideBordersConfig,
  value?: SideBordersConfig | BorderProperties,
): SideBorders => {
  if (typeof value === 'undefined') {
    return configToSides<BorderProperties>(defaultSideBorders, defaults)
  }
  const defs = extract(defaults)
  const defWidths = configToSides<Size>(defaultBorderWidths, defs.widths)
  const defStyles = configToSides<BorderStyle>(defaultBorderStyles, defs.styles)
  const props = extract(value)
  const widths = configToSides<Size>(defWidths, props.widths)
  const styles = configToSides<BorderStyle>(defStyles, props.styles)
  return {
    top: {
      width: widths.top,
      style: styles.top,
    },
    bottom: {
      width: widths.bottom,
      style: styles.bottom,
    },
    left: {
      width: widths.left,
      style: styles.left,
    },
    right: {
      width: widths.right,
      style: styles.right,
    },
  }
}
