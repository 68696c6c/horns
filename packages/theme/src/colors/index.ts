/* istanbul ignore file */
export { Colors, makeColors } from './colors'
export { Config as ColorsConfig } from './config'
export { Colorway, ColorwayNotification } from './types'
