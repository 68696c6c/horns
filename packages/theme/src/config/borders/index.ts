export {
  BorderStyle,
  evalSideBordersConfigs,
  SideBorders,
  SideBordersConfig,
} from './borders'
