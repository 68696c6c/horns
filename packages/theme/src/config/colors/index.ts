export { Colors, makeColors } from './colors'
export { Config as ColorsConfig } from './config'
export { Color } from './types'
