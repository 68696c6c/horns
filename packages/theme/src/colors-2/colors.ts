import { Colorways, makeColorways } from './colorways'
import { Color, Swatches } from './types'
import { Config } from './config'
import { ColorShades } from './shades'

export type Colors = {
  [key in Color]: Swatches
}

export const makeColors = (
  config: Config,
  shades: ColorShades,
): {
  colors: Colors
  colorways: Colorways
} => {
  const colorways = makeColorways(config, shades)
  return {
    colors: {
      primary: colorways.primary.base,
      secondary: colorways.secondary.base,
      tertiary: colorways.tertiary.base,
      prominent: colorways.prominent.base,
      selected: colorways.selected.base,
      action: colorways.action.base,
      success: colorways.success.base,
      info: colorways.info.base,
      warning: colorways.warning.base,
      danger: colorways.danger.base,
      dark: colorways.dark.base,
      neutral: colorways.neutral.base,
      light: colorways.light.base,
      background: colorways.background.base,
      backgroundAlt: colorways.backgroundAlt.base,
      typography: colorways.typography.base,
    },
    colorways,
  }
}
