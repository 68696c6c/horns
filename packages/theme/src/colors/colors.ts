import { mergeConfig } from '../utils'

import {
  ColorwayStates,
  makeBgPrimaryColorway,
  makeColorway,
  makeDarkColorway,
  makeLightColorway,
  makeBgSecondaryColorway,
  makeBgInverseColorway,
} from './colorways'
import { defaultConfig, Config } from './config'
import { makeColorShades } from './shades'
import { BaseBrandColor, BaseColor, Color, Mode } from './types'
import DeepPartial from '../utils/deep-partial'

export type BrandColorways = {
  [key in BaseBrandColor]: ColorwayStates
}

export const determineBrandColorway = (
  input: BaseColor,
  { primary, secondary, tertiary }: BrandColorways,
): ColorwayStates => {
  switch (input) {
    default:
      return primary
    case Color.Secondary:
      return secondary
    case Color.Tertiary:
      return tertiary
  }
}

export type Colors = Required<
  {
    [key in Color]: ColorwayStates
  }
>

export const makeColors = (config?: DeepPartial<Config>): Colors => {
  const mergedConfig = mergeConfig<Config>(defaultConfig, config)
  const shades = makeColorShades(mergedConfig)
  const isDark = mergedConfig.mode === Mode.Dark
  const brandColorways: BrandColorways = {
    [Color.Primary]: makeColorway(shades.primary, isDark),
    [Color.Secondary]: makeColorway(shades.secondary, isDark),
    [Color.Tertiary]: makeColorway(shades.tertiary, isDark),
  }
  return {
    ...brandColorways,
    [Color.Action]: determineBrandColorway(mergedConfig.action, brandColorways),
    [Color.Prominent]: determineBrandColorway(
      mergedConfig.prominent,
      brandColorways,
    ),
    [Color.Selected]: determineBrandColorway(
      mergedConfig.selected,
      brandColorways,
    ),
    [Color.Success]: makeColorway(shades.success, isDark),
    [Color.Info]: makeColorway(shades.info, isDark),
    [Color.Warning]: makeColorway(shades.warning, isDark),
    [Color.Danger]: makeColorway(shades.danger, isDark),
    [Color.Dark]: makeDarkColorway(shades.dark),
    [Color.Neutral]: makeColorway(shades.neutral, isDark),
    [Color.Light]: makeLightColorway(shades.light),
    [Color.BgPrimary]: makeBgPrimaryColorway(shades, isDark),
    [Color.BgSecondary]: makeBgSecondaryColorway(shades, isDark),
    [Color.BgInverse]: makeBgInverseColorway(shades, isDark),
  }
}
