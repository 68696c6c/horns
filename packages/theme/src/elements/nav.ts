import { BorderStyle, evalSideBordersConfigs } from '../borders'
import { Color } from '../colors'
import { ShadowType } from '../shadows'
import { Size, evalCornerSizesConfigs, evalSideSizesConfigs } from '../sizes'
import { Font } from '../typography'
import { Cursor } from '../utils'

import {
  ElementConfig,
  ElementProps,
  ThemeElement,
  evalBooleanProp,
} from './elements'

type NavItemConfig = Omit<
  ElementConfig,
  'outlined' | 'shadowed' | 'shadowType' | 'typographic'
>

export interface NavConfig extends ElementConfig {
  currentItem?: NavItemConfig
}

export interface NavProps extends ElementProps {
  currentItem?: ElementProps
}

export interface NavTheme extends ThemeElement {
  currentItem: ThemeElement
}

const defaultCurrentItem: NavItemConfig = {
  border: {
    bottom: {
      width: Size.Tiny,
      style: BorderStyle.Solid,
    },
  },
  color: Color.Selected,
  cursor: Cursor.Default,
  font: Font.Compact,
  padding: {
    x: Size.Medium,
    y: Size.Small,
  },
  radius: {
    all: Size.None,
  },
}

export const defaultNav: NavConfig = {
  border: {
    all: {
      width: Size.None,
      style: BorderStyle.None,
    },
  },
  color: Color.BgPrimary,
  cursor: Cursor.Pointer,
  font: Font.Compact,
  outlined: false,
  padding: {
    x: Size.Medium,
    y: Size.Small,
  },
  radius: {
    all: Size.None,
  },
  shadowed: false,
  shadowType: ShadowType.Box,
  typographic: false,
  currentItem: defaultCurrentItem,
}

export const makeNav = (input?: NavProps): NavTheme => {
  const border = evalSideBordersConfigs(defaultNav.border, input?.border)
  const outlined = evalBooleanProp(defaultNav.outlined, 'outlined', input)
  const padding = evalSideSizesConfigs(defaultNav.padding, input?.padding)
  const radius = evalCornerSizesConfigs(defaultNav.radius, input?.radius)
  const shadow = input?.shadowed ? defaultNav.shadowType : ShadowType.None
  const { typographic } = defaultNav

  const item = input?.currentItem
  const itemPadding = evalSideSizesConfigs(padding, item?.padding)
  const itemRadius = evalCornerSizesConfigs(radius, item?.radius)

  return {
    border,
    color: input?.color || defaultNav.color,
    cursor: input?.cursor || defaultNav.cursor,
    font: input?.font || defaultNav.font,
    outlined,
    padding,
    shadow,
    radius,
    typographic,
    currentItem: {
      border: evalSideBordersConfigs(
        defaultNav.border,
        defaultCurrentItem.border,
        input?.border,
        item?.border,
      ),
      color: item?.color || input?.color || defaultCurrentItem.color,
      cursor: item?.cursor || input?.cursor || defaultCurrentItem.cursor,
      font: item?.font || input?.font || defaultCurrentItem.font,
      outlined,
      padding: evalSideSizesConfigs(defaultCurrentItem.padding, itemPadding),
      shadow,
      radius: evalCornerSizesConfigs(defaultCurrentItem.radius, itemRadius),
      typographic,
    },
  }
}
