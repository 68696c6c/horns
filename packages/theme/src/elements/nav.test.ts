import { BorderStyle } from '../borders'
import { Colorway } from '../colors'
import { Cursor } from '../cursors'
import { Size } from '../sizes'
import { Font } from '../typography'

import { defaultNav, makeNav, NavConfig } from './nav'

describe('makeNav', () => {
  const expected = makeNav(defaultNav)

  it('should return default values if no input is provided', () => {
    const result = makeNav()
    expect(result).toStrictEqual(expected)
  })

  it('should use default values for incomplete inputs', () => {
    const input: Partial<NavConfig> = {
      color: Colorway.Tertiary,
      currentItem: { color: Colorway.Secondary },
    }
    const result = makeNav(input)
    expect(result).toMatchSnapshot()
  })

  describe('border', () => {
    it('should accept "all" inputs', () => {
      const navBorder = {
        style: BorderStyle.Groove,
        width: Size.Giant,
      }
      const itemBorder = {
        style: BorderStyle.Double,
        width: Size.Medium,
      }
      const input: Partial<NavConfig> = {
        border: {
          all: navBorder,
        },
        currentItem: {
          border: {
            all: itemBorder,
          },
        },
      }
      const result = makeNav(input)
      expect(result.border).toStrictEqual({
        top: navBorder,
        bottom: navBorder,
        left: navBorder,
        right: navBorder,
      })
      expect(result.currentItem.border).toStrictEqual({
        top: itemBorder,
        bottom: itemBorder,
        left: itemBorder,
        right: itemBorder,
      })
    })

    it('should accept dimensional inputs', () => {
      const input: Partial<NavConfig> = {
        border: {
          x: {
            style: BorderStyle.Groove,
            width: Size.Giant,
          },
          y: {
            style: BorderStyle.Double,
            width: Size.XXSmall,
          },
        },
        currentItem: {
          border: {
            x: {
              style: BorderStyle.Dotted,
              width: Size.Medium,
            },
            y: {
              style: BorderStyle.Dashed,
              width: Size.XLarge,
            },
          },
        },
      }
      const result = makeNav(input)
      expect(result.border).toStrictEqual({
        top: {
          style: BorderStyle.Double,
          width: Size.XXSmall,
        },
        bottom: {
          style: BorderStyle.Double,
          width: Size.XXSmall,
        },
        left: {
          style: BorderStyle.Groove,
          width: Size.Giant,
        },
        right: {
          style: BorderStyle.Groove,
          width: Size.Giant,
        },
      })
      expect(result.currentItem.border).toStrictEqual({
        top: {
          style: BorderStyle.Dashed,
          width: Size.XLarge,
        },
        bottom: {
          style: BorderStyle.Dashed,
          width: Size.XLarge,
        },
        left: {
          style: BorderStyle.Dotted,
          width: Size.Medium,
        },
        right: {
          style: BorderStyle.Dotted,
          width: Size.Medium,
        },
      })
    })

    it('should accept complete inputs', () => {
      const input: Partial<NavConfig> = {
        border: {
          top: {
            style: BorderStyle.Double,
            width: Size.Giant,
          },
          bottom: {
            style: BorderStyle.Dashed,
            width: Size.XXLarge,
          },
          left: {
            style: BorderStyle.Groove,
            width: Size.XLarge,
          },
          right: {
            style: BorderStyle.Dotted,
            width: Size.Medium,
          },
        },
        currentItem: {
          border: {
            top: {
              style: BorderStyle.Inset,
              width: Size.Small,
            },
            bottom: {
              style: BorderStyle.Outset,
              width: Size.XSmall,
            },
            left: {
              style: BorderStyle.Ridge,
              width: Size.XXSmall,
            },
            right: {
              style: BorderStyle.Hidden,
              width: Size.Tiny,
            },
          },
        },
      }
      const result = makeNav(input)
      expect(result.border).toStrictEqual({
        top: {
          style: BorderStyle.Double,
          width: Size.Giant,
        },
        bottom: {
          style: BorderStyle.Dashed,
          width: Size.XXLarge,
        },
        left: {
          style: BorderStyle.Groove,
          width: Size.XLarge,
        },
        right: {
          style: BorderStyle.Dotted,
          width: Size.Medium,
        },
      })
      expect(result.currentItem.border).toStrictEqual({
        top: {
          style: BorderStyle.Inset,
          width: Size.Small,
        },
        bottom: {
          style: BorderStyle.Outset,
          width: Size.XSmall,
        },
        left: {
          style: BorderStyle.Ridge,
          width: Size.XXSmall,
        },
        right: {
          style: BorderStyle.Hidden,
          width: Size.Tiny,
        },
      })
    })

    it('should merge nav and item inputs', () => {
      const input: Partial<NavConfig> = {
        border: {
          all: {
            style: BorderStyle.Double,
            width: Size.Giant,
          },
        },
        currentItem: {
          border: {
            top: {
              style: BorderStyle.Outset,
              width: Size.XSmall,
            },
          },
        },
      }
      const result = makeNav(input)
      expect(result.border).toStrictEqual({
        top: {
          style: BorderStyle.Double,
          width: Size.Giant,
        },
        bottom: {
          style: BorderStyle.Double,
          width: Size.Giant,
        },
        left: {
          style: BorderStyle.Double,
          width: Size.Giant,
        },
        right: {
          style: BorderStyle.Double,
          width: Size.Giant,
        },
      })
      expect(result.currentItem.border).toStrictEqual({
        top: {
          style: BorderStyle.Outset,
          width: Size.XSmall,
        },
        bottom: {
          style: BorderStyle.Double,
          width: Size.Giant,
        },
        left: {
          style: BorderStyle.Double,
          width: Size.Giant,
        },
        right: {
          style: BorderStyle.Double,
          width: Size.Giant,
        },
      })
    })
  })

  it('should accept color values', () => {
    const input: Partial<NavConfig> = {
      color: Colorway.Tertiary,
      currentItem: { color: Colorway.Secondary },
    }
    const result = makeNav(input)
    expect(result.color).toBe(Colorway.Tertiary)
    expect(result.currentItem.color).toBe(Colorway.Secondary)
  })

  it('should accept cursor values', () => {
    const input: Partial<NavConfig> = {
      cursor: Cursor.Pointer,
      currentItem: { cursor: Cursor.Progress },
    }
    const result = makeNav(input)
    expect(result.cursor).toBe(Cursor.Pointer)
    expect(result.currentItem.cursor).toBe(Cursor.Progress)
  })

  it('should accept font values', () => {
    const input: Partial<NavConfig> = {
      font: Font.Control,
      currentItem: { font: Font.Control },
    }
    const result = makeNav(input)
    expect(result.font).toBe(Font.Control)
    expect(result.currentItem.font).toBe(Font.Control)
  })

  describe('padding', () => {
    it('should accept "all" inputs', () => {
      const input: Partial<NavConfig> = {
        padding: { all: Size.Giant },
        currentItem: {
          padding: { all: Size.Medium },
        },
      }
      const result = makeNav(input)
      expect(result.padding).toStrictEqual({
        top: Size.Giant,
        bottom: Size.Giant,
        left: Size.Giant,
        right: Size.Giant,
      })
      expect(result.currentItem.padding).toStrictEqual({
        top: Size.Medium,
        bottom: Size.Medium,
        left: Size.Medium,
        right: Size.Medium,
      })
    })

    it('should accept dimensional inputs', () => {
      const input: Partial<NavConfig> = {
        padding: {
          x: Size.Giant,
          y: Size.Large,
        },
        currentItem: {
          padding: {
            x: Size.Small,
            y: Size.Medium,
          },
        },
      }
      const result = makeNav(input)
      expect(result.padding).toStrictEqual({
        top: Size.Large,
        bottom: Size.Large,
        left: Size.Giant,
        right: Size.Giant,
      })
      expect(result.currentItem.padding).toStrictEqual({
        top: Size.Medium,
        bottom: Size.Medium,
        left: Size.Small,
        right: Size.Small,
      })
    })

    it('should accept complete inputs', () => {
      const input: Partial<NavConfig> = {
        padding: {
          top: Size.Giant,
          bottom: Size.Large,
          left: Size.Medium,
          right: Size.Small,
        },
        currentItem: {
          padding: {
            top: Size.XXLarge,
            bottom: Size.XLarge,
            left: Size.XSmall,
            right: Size.XXSmall,
          },
        },
      }
      const result = makeNav(input)
      expect(result.padding).toStrictEqual({
        top: Size.Giant,
        bottom: Size.Large,
        left: Size.Medium,
        right: Size.Small,
      })
      expect(result.currentItem.padding).toStrictEqual({
        top: Size.XXLarge,
        bottom: Size.XLarge,
        left: Size.XSmall,
        right: Size.XXSmall,
      })
    })

    it('should merge nav and item inputs', () => {
      const input: Partial<NavConfig> = {
        padding: { all: Size.Giant },
        currentItem: {
          padding: { top: Size.XXSmall },
        },
      }
      const result = makeNav(input)
      expect(result.padding).toStrictEqual({
        top: Size.Giant,
        bottom: Size.Giant,
        left: Size.Giant,
        right: Size.Giant,
      })
      expect(result.currentItem.padding).toStrictEqual({
        top: Size.XXSmall,
        bottom: Size.Giant,
        left: Size.Giant,
        right: Size.Giant,
      })
    })
  })

  describe('radius', () => {
    it('should accept "all" inputs', () => {
      const input: Partial<NavConfig> = {
        radius: { all: Size.Giant },
        currentItem: {
          radius: { all: Size.Medium },
        },
      }
      const result = makeNav(input)
      expect(result.radius).toStrictEqual({
        topRight: Size.Giant,
        bottomRight: Size.Giant,
        topLeft: Size.Giant,
        bottomLeft: Size.Giant,
      })
      expect(result.currentItem.radius).toStrictEqual({
        topRight: Size.Medium,
        bottomRight: Size.Medium,
        topLeft: Size.Medium,
        bottomLeft: Size.Medium,
      })
    })

    it('should accept dimensional inputs', () => {
      const input: Partial<NavConfig> = {
        radius: {
          left: Size.Giant,
          right: Size.Large,
        },
        currentItem: {
          radius: {
            top: Size.Small,
            bottom: Size.Medium,
          },
        },
      }
      const result = makeNav(input)
      expect(result.radius).toStrictEqual({
        topRight: Size.Large,
        bottomRight: Size.Large,
        topLeft: Size.Giant,
        bottomLeft: Size.Giant,
      })
      expect(result.currentItem.radius).toStrictEqual({
        topRight: Size.Small,
        bottomRight: Size.Medium,
        topLeft: Size.Small,
        bottomLeft: Size.Medium,
      })
    })

    it('should accept complete inputs', () => {
      const input: Partial<NavConfig> = {
        radius: {
          topRight: Size.Giant,
          bottomRight: Size.Large,
          topLeft: Size.Medium,
          bottomLeft: Size.Small,
        },
        currentItem: {
          radius: {
            topRight: Size.XXLarge,
            bottomRight: Size.XLarge,
            topLeft: Size.XSmall,
            bottomLeft: Size.XXSmall,
          },
        },
      }
      const result = makeNav(input)
      expect(result.radius).toStrictEqual({
        topRight: Size.Giant,
        bottomRight: Size.Large,
        topLeft: Size.Medium,
        bottomLeft: Size.Small,
      })
      expect(result.currentItem.radius).toStrictEqual({
        topRight: Size.XXLarge,
        bottomRight: Size.XLarge,
        topLeft: Size.XSmall,
        bottomLeft: Size.XXSmall,
      })
    })

    it('should merge nav and item inputs', () => {
      const input: Partial<NavConfig> = {
        radius: { all: Size.Giant },
        currentItem: {
          radius: { topLeft: Size.XXSmall },
        },
      }
      const result = makeNav(input)
      expect(result.radius).toStrictEqual({
        topRight: Size.Giant,
        bottomRight: Size.Giant,
        topLeft: Size.Giant,
        bottomLeft: Size.Giant,
      })
      expect(result.currentItem.radius).toStrictEqual({
        topRight: Size.Giant,
        bottomRight: Size.Giant,
        topLeft: Size.XXSmall,
        bottomLeft: Size.Giant,
      })
    })
  })
})
