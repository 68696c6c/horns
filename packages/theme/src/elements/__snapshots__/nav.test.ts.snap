// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`makeNav should use default values for incomplete inputs 1`] = `
Object {
  "border": Object {
    "bottom": Object {
      "style": "none",
      "width": "none",
    },
    "left": Object {
      "style": "none",
      "width": "none",
    },
    "right": Object {
      "style": "none",
      "width": "none",
    },
    "top": Object {
      "style": "none",
      "width": "none",
    },
  },
  "color": "tertiary",
  "currentItem": Object {
    "border": Object {
      "bottom": Object {
        "style": "solid",
        "width": "tiny",
      },
      "left": Object {
        "style": "none",
        "width": "none",
      },
      "right": Object {
        "style": "none",
        "width": "none",
      },
      "top": Object {
        "style": "none",
        "width": "none",
      },
    },
    "color": "secondary",
    "cursor": "default",
    "font": "compact",
    "outlined": false,
    "padding": Object {
      "bottom": "small",
      "left": "medium",
      "right": "medium",
      "top": "small",
    },
    "radius": Object {
      "bottomLeft": "none",
      "bottomRight": "none",
      "topLeft": "none",
      "topRight": "none",
    },
    "shadow": "none",
    "typographic": false,
  },
  "cursor": "pointer",
  "font": "compact",
  "outlined": false,
  "padding": Object {
    "bottom": "small",
    "left": "medium",
    "right": "medium",
    "top": "small",
  },
  "radius": Object {
    "bottomLeft": "none",
    "bottomRight": "none",
    "topLeft": "none",
    "topRight": "none",
  },
  "shadow": "none",
  "typographic": false,
}
`;
